prometheus-exporter/script/ConfigArgs.py

```diff
@@ -0,0 +1,73 @@
+import yaml
+import re
+
+class ConfigArgs:
+    def __init__(self, yaml_config: dict):
+        self.prometheus_url = self.validate_url(yaml_config["prometheus_url"])
+        self.start_date = self.validate_int("start_date", yaml_config["start_date"])
+        self.end_date = self.validate_int("end_date", yaml_config["end_date"])
+        self.step = self.validate_int("interval", yaml_config["interval"])
+        self.generate_graphs = self.validate_bool("generate_graphs", yaml_config["generate_graphs"])
+        self.authentication = self.validate_auth(yaml_config["authentication"])
+        self.metrics = self.validate_metrics(yaml_config["metrics"])
+    
+    def validate_url(self, arg_value, pat=re.compile(r"^https?://.+$")):
+        if not pat.match(arg_value):
+            raise yaml.YAMLError(f"prometheus_url: Invalid url value {arg_value}")
+        return arg_value
+    
+    def validate_int(self, arg, value):
+        if not isinstance(value, int):
+            raise yaml.YAMLError(f"{arg}: Expected an integer, but got {type(value).__name__} instead.")
+        return value
+    
+    def validate_bool(self, arg, value):
+        if not isinstance(value, bool):
+            raise yaml.YAMLError(f"{arg}: Expected a boolean, but got {type(value).__name__} instead.")
+        return value
+    
+    def validate_user_pass(self, value):
+        if len(value.split(':')) != 2:
+            raise yaml.YAMLError(f"authentication: user_password must be in the format 'user:password'")
```

Comment:
Si es un string sin variables interpoladas no hace falta que sea un f-string, por temas de estilo y constancia convertiría los f-strings que no necesitan serlo a strings normales
```suggestion
            raise yaml.YAMLError("authentication: user_password must be in the format 'user:password'")
```

Reply:
corregido